### DISCUSSION TITLE
Better Documentation

### DISCUSSION THREAD

**Discussion by EfstathiadisD** [chakra-ui/ark#2127] [Ideas]
I have found that when Developing components with Ark UI I need to visit also Zag. 

The most important reason is [data-attributes]. 

But beyond that I find that asChild forwarding isn't as straightforward for some components.

One idea is to port some of Zags Docs to Ark UI as well or Link to the corresponding Zag component. 

I think that people don't know or understand that, since Ark is quite new and it needs to at least be pointed out cause it is not representative of how good the Lib is.

Of course that's the lowest hanging fruit and would love to see an even better approach but this Discussion is meant to foster that. Thank you!🙏

**Comment by Omikorin** [comment]
Thanks, we are working on this #2121 

Could you explain a bit further about asChild issue?

**Reply by EfstathiadisD** [reply]
Nice. Like it! Re. the asChild there are some instances when you try to use it with other Libs (i.e: Joy UI or React Aria) which are nοτ working as expected. What doesn't work is prob the propagation of some props. That's not the problem. The issue with the Docs there is it doen't give clues how to identify what isn't working.. 2 Examples..

```tsx
import { forwardRef, type ReactNode } from "react";

import { PinInput, type PinInputRootProps } from "@ark-ui/react";

import FormLabel from "@mui/joy/FormLabel";

import { cn } from "#src/utils/cn";

type PinProps = PinInputRootProps & {
  label: ReactNode;
};

export const Pin = forwardRef<HTMLDivElement, PinProps>(
  ({ label, className, ...props }, ref) => {
    return (
      <PinInput.Root
        autoFocus
        className={cn("flex flex-col", className)}
        ref={ref}
        {...props}
      >
        <PinInput.Label asChild>
          <FormLabel>{label}</FormLabel>
        </PinInput.Label>
        <PinInput.Control className="grid grid-cols-6 gap-3">
          {Array.from({ length: 6 }, (_, index) => index).map((id, index) => (
            <PinInput.Input
              className="h-9 w-full rounded-sm border border-solid border-neutral-300 bg-transparent text-center shadow-none outline-none focus:outline focus:-outline-offset-1 focus:outline-primary-500 dark:border-neutral-600 dark:focus:outline dark:focus:-outline-offset-1 dark:focus:outline-primary-600"
              index={index}
              key={id}
            />
          ))}
        </PinInput.Control>
      </PinInput.Root>
    );
  },
);

Pin.displayName = "Pin";
```
This is a working Pin Input customized to the needs of our app. But when I use the Input from Joy UI, it doesn't fully work. (Doesn't go to the next input when I enter a number and Past is a bit weird). That's OK and it prob can be fixed but I would like to understand what is wrong. Cause now I have no idea.

The same issue exist with Joy UI "FormControl" and DatePicker component.. Smth isn't propagating.. The Docs should point somehow to a resolution method. That's all. Thanks for the reply..

**Reply by Omikorin** [reply]
I'm not sure about this. @cschroeter what do you think?

**Reply by cschroeter** [reply]
@EfstathiadisD 

Could you instead of mapping over the array try to write it out:

```jsx
<PinInput.Control>
  <PinInput.Input asChild>
    <Input />
  </PinInput.Input>
  {/* etc. */}
</PinInput.Control>
```

**Reply by cschroeter** [reply]
@EfstathiadisD 

You may want to compare your current approach with https://park-ui.com/docs/panda/components/pin-input

**Reply by EfstathiadisD** [reply]
Will try it out and let you know. Thanks

**Reply by EfstathiadisD** [reply]
Hello. Sorry it took so much. Tried this weekend with the latest and implemented them 1-2-1 but still not working. The approaches are pretty similar. Joy UI has a regular ref so it should be propagated but it doesn't. Will let you folks know if I figure it out.. Thanks!
